fdbclient/DatabaseContext.h

```diff
@@ -611,6 +611,14 @@ class DatabaseContext : public ReferenceCounted<DatabaseContext>, public FastAll
 	// Cache of the latest commit versions of storage servers.
 	VersionVector ssVersionVectorCache;
 
+	// Introduced mainly to optimize out the version vector related code (on the client side)
```

Comment:
Shall we propagate the version vector knob enable/disable information, from GRV proxy to client, in "GetReadVersionReply" message and use that information in place of using this check? But I think that idea may not work correctly on recovery boundaries (because "GetReadVersionReply" messages may cross recovery boundaries). We can explore that idea further if we think that is a better approach than using this check.

Reply:
- it seems returning vv status from the GRV request would be better- Does the client know the epoch changed or the GRV is less than the new recovery version?
- I'd prefer removing `mayNeed` from the function name. It sounds ambiguous.. perhaps rename to `versionVectorCacheActive()` 
- can you make this an inline function for speed

Reply:
Done.

Reply:
> it seems returning vv status from the GRV request would be better-

I thought more about it. This might not work if the knob was turned on before recovery and turned off after that. The version vector cache won't get cleared after recovery and reads could reference a stale latest commit version from the version vector.